Docs 7.1.1 JAX compatibility pages (#5708)
RadeonOpenCompute/ROCm: docs/compatibility/ml-compatibility/jax-compatibility.rst

  Refer to the official Jax documentation for the most up-to-date information.

Key features and enhancements for ROCm 7.1
===============================================================================

- Enabled compilation of multihost HLO runner Python bindings.

  - Backported multihost HLO runner bindings and some related changes to
    :code:`FunctionalHloRunner`.

  - Added :code:`requirements_lock_3_12` to enable building for Python 3.12.

- Removed hardcoded NHWC convolution layout for ``fp16`` precision to address the performance drops for ``fp16`` precision on gfx12xx GPUs.

- ROCprofiler-SDK integration:

  - Integrated ROCprofiler-SDK (v3) to XLA to improve profiling of GPU events,
    support both time-based and step-based profiling.

  - Added unit tests for :code:`rocm_collector` and :code:`rocm_tracer`.

- Added Triton unsupported conversion from ``f8E4M3FNUZ`` to ``fp16`` with
  rounding mode.

- Introduced :code:`CudnnFusedConvDecomposer` to revert fused convolutions
  when :code:`ConvAlgorithmPicker` fails to find a fused algorithm, and removed
  unfused fallback paths from :code:`RocmFusedConvRunner`.

> Key features and enhancements for ROCm 7.0

Reviewer: is this line required? already there at line 272..

  JAX API modules are maintained by the JAX project and is subject to change.
  Refer to the official Jax documentation for the most up-to-date information.

Key features and enhancements for ROCm 7.1
===============================================================================

> - Enable compilation of multihost HLO runner python bindings

Reviewer 2: ```suggestion
- Enabled compilation of multihost HLO runner Python bindings.
```

>   - Backport multihost HLO runner bindings and some related changes to

Reviewer 2: ```suggestion
  - Backported multihost HLO runner bindings and some related changes to
```

>     :code:`FunctionalHloRunner`

Reviewer 2: ```suggestion
    :code:`FunctionalHloRunner`.
```

>   - Add :code:`requirements_lock_3_12` to enable building for python 3.12.

Reviewer 2: ```suggestion
  - Added :code:`requirements_lock_3_12` to enable building for Python 3.12.
```

> - Performance drops for ``fp16`` precision on gfx12xx GPUs were observed.

Reviewer 2: ```suggestion
- Removed hardcoded NHWC convolution layout for ``fp16`` precision to address the performance drops for ``fp16`` precision on gfx12xx GPUs.
```

>   - Remove hardcoded NHWC convolution layout for ``fp16`` precision.

Reviewer 2: ```suggestion
```

- ROCprofiler-SDK integration:

>   - Integrating ROCprofiler-SDK (v3) to XLA to improve profiling of GPU events,

Reviewer 2: ```suggestion
  - Integrated ROCprofiler-SDK (v3) to XLA to improve profiling of GPU events,
```

    support both time-based and step-based profiling.

>   - Add unit tests for :code:`rocm_collector` and :code:`rocm_tracer`.

Reviewer 2: ```suggestion
  - Added unit tests for :code:`rocm_collector` and :code:`rocm_tracer`.
```

- Added Triton unsupported conversion from ``f8E4M3FNUZ`` to ``fp16`` with
  rounding mode.

> - Introduce :code:`CudnnFusedConvDecomposer` to revert back fused convs if no

Reviewer 2: ```suggestion
- Introduced :code:`CudnnFusedConvDecomposer` to revert fused convs if no
```

Author: Fixed with commit, please check I did minor changes.

>   fused algorithm could be found with :code:`ConvAlgorithmPicker`. Remove

Reviewer 2: ```suggestion
  fused algorithm could be found with :code:`ConvAlgorithmPicker`. Removed
```

Author: Fixed with commit, please check I did minor changes.
